twoliter/src/lock.rs

```diff
@@ -198,42 +172,41 @@ struct OCIArchive {
     image: LockedImage,
     digest: String,
     cache_dir: PathBuf,
+    temp: TempDir,
 }
 
 impl OCIArchive {
-    fn new<P>(image: &LockedImage, digest: &str, cache_dir: P) -> Self
+    fn new<P>(image: &LockedImage, digest: &str, cache_dir: P) -> Result<Self>
     where
         P: AsRef<Path>,
     {
-        Self {
+        Ok(Self {
             image: image.clone(),
             digest: digest.into(),
             cache_dir: cache_dir.as_ref().to_path_buf(),
-        }
+            temp: TempDir::new_in(cache_dir.as_ref()).context(format!(
+                "failed to create temporary directory in {}",
+                cache_dir.as_ref().display()
+            ))?,
```

Comment:
This temp dir bothers me for a couple of reasons. I don't think it's very well-named since it ends up being the primary way that blobs and the manifest are passed back to the caller.

The docker treatment is a bit strange, you end up creating another temp dir inside the cache dir, where the contents are unpacked to the provided dir, which is this temp dir passed from the caller.

The crane CLI doesn't use the cache dir at all and just populates the provided dir, which is this temp dir inside the cache dir.

I would try to reduce this to just one directory that gets passed around, and maybe `cache_dir` can be `work_dir` - a temporary directory that's created by the caller that instantiates the `OCIArchive`, on the same filesystem as the eventual output directory, with the contract "it is OK to write into this directory, including creating any temporary dirs needed, but you're expected to at least create a `index.json` and some files under `blobs`.

Reply:
Not sure if my change does what you had intended, but I realized that the two directories had no real purpose anymore. So i kept it to one and actually removed it as a tempdir on the OCIArchive side because turns out using the temp dir prevented that digest caching check from even working..woops. Now only the docker approach creates a temp directory for saving the archive out so that the appropriate cache directory is created